eth/beacon/db/chain.py

```diff
@@ -0,0 +1,450 @@
+
+    @staticmethod
+    def _get_block_by_hash(db: BaseDB, block_hash: Hash32) -> BaseBeaconBlock:
+        """
+        Returns the requested block header as specified by block hash.
+
+        Raises BlockNotFound if it is not present in the db.
+
+        NOTE: copied from ``eth.db.header.HeaderDB._get_block_by_hash``,
+        and changed "header" to "block".
+        """
+        validate_word(block_hash, title="Block Hash")
+        try:
+            block_rlp = db[block_hash]
+        except KeyError:
+            raise BlockNotFound("No block with hash {0} found".format(
+                encode_hex(block_hash)))
+        return _decode_block(block_rlp)
+
+    def get_score(self, block_hash: Hash32) -> int:
+        """
+        NOTE: copied from ``eth.db.header.HeaderDB.get_score``,
+        and changed "header" to "block".
+        """
+        return self._get_score(self.db, block_hash)
+
+    @staticmethod
+    def _get_score(db: BaseDB, block_hash: Hash32) -> int:
+        """
+        NOTE: copied from ``eth.db.header.HeaderDB._get_score``,
+        and changed "header" to "block".
+        """
+        try:
+            encoded_score = db[SchemaV1.make_block_hash_to_score_lookup_key(block_hash)]
+        except KeyError:
+            raise BlockNotFound("No block with hash {0} found".format(
+                encode_hex(block_hash)))
+        return rlp.decode(encoded_score, sedes=rlp.sedes.big_endian_int)
+
+    def block_exists(self, block_hash: Hash32) -> bool:
+        """
+        NOTE: copied from ``eth.db.header.HeaderDB.header_exists``,
+        and changed "header" to "block".
+        """
+        return self._block_exists(self.db, block_hash)
+
+    @staticmethod
+    def _block_exists(db: BaseDB, block_hash: Hash32) -> bool:
+        """
+        NOTE: copied from ``eth.db.header.HeaderDB._header_exists``,
+        and changed "header" to "block".
+        """
+        validate_word(block_hash, title="Block Hash")
+        return block_hash in db
+
+    def persist_block_chain(
+        self,
+        headers: Iterable[BaseBeaconBlock]
+    ) -> Tuple[Tuple[BaseBeaconBlock, ...], Tuple[BaseBeaconBlock, ...]]:
+        """
+        Return two iterable of blocks, the first containing the new canonical blocks,
+        the second containing the old canonical headers
+
+        NOTE: copied from ``eth.db.header.HeaderDB.persist_header_chain``,
+        and changed "header" to "block".
+        """
+        with self.db.atomic_batch() as db:
+            return self._persist_block_chain(db, headers)
+
+    @classmethod
+    def _persist_block_chain(
+            cls,
+            db: BaseDB,
+            blocks: Iterable[BaseBeaconBlock]
+    ) -> Tuple[Tuple[BaseBeaconBlock, ...], Tuple[BaseBeaconBlock, ...]]:
+        """
+
+        NOTE: copied from ``eth.db.header.HeaderDB._persist_header_chain``,
+        and changed "header" to "block".
+        NOTE: moodified a lot
```

Comment:
Probably worth dropping these comments in cases where the modification sufficiently departs from the original one...

Reply:
👍 these comments were just for me to track the source in case you want me to deduplicate the code.